### PR TITLE
Makes compute node a SameServer entity if it has multiple children

### PR DESCRIPTION
**Pull request by robertgmoss**

### CODE REVIEW

File: brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaPlanToSpecTransformer.java [hunk 1]
```diff
@@ -184,79 +190,118 @@ private static boolean isTosca(Map<?,?> obj) {
...
-        Requirement host = nodeTemplate.getRequirements() != null ? nodeTemplate.getRequirements().get("host") : null;
-        if (host != null) {
-            for (RelationshipTemplate r : nodeTemplate.getRelationships().values()) {
-                if (r.getRequirementName().equals("host")) {
-                    return r.getTarget();
+    public static class DependencyTree{
+
+        private Map<String, String> roots = MutableMap.of();
+        private Map<String, String> parents = MutableMap.of();
+        private Map<String, List<String>> children = MutableMap.of();
```
Comment (sjcorbett) [inline brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaPlanToSpecTransformer.java]: Use a [MultiMap](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Multimap.html) for this field.

File: brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaPlanToSpecTransformer.java [hunk 2]
```diff
@@ -184,79 +190,118 @@ private static boolean isTosca(Map<?,?> obj) {
...
+                            nodeTemplate.getType(), topo.getDependencies());
+
+            EntitySpec<? extends Entity> spec = ToscaNodeToEntityConverter.with(mgmt)
+                    .setNodeId(root)
+                    .setNodeTemplate(nodeTemplate)
+                    .setIndexedNodeTemplate(indexedNodeTemplate)
+                    .setCsarFileRepository(csarFileRepository)
+                    .createSpec(hasMultipleChildren(root));
+
+            for(String child : children.getOrDefault(root, Collections.<String>emptyList())) {
```
Comment (sjcorbett) [inline brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaPlanToSpecTransformer.java]: `getOrDefault` is unavailable in Java 1.7.

File: brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaPlanToSpecTransformer.java [hunk 3]
```diff
@@ -184,79 +190,118 @@ private static boolean isTosca(Map<?,?> obj) {
         return createApplicationSpec(application.getName(), dt, id);
     }
 
-    private String getParentId(NodeTemplate nodeTemplate) {
-        Requirement host = nodeTemplate.getRequirements() != null ? nodeTemplate.getRequirements().get("host") : null;
-        if (host != null) {
-            for (RelationshipTemplate r : nodeTemplate.getRelationships().values()) {
-                if (r.getRequirementName().equals("host")) {
-                    return r.getTarget();
+    public static class DependencyTree{
```
Comment (sjcorbett) [inline brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaPlanToSpecTransformer.java]: Interesting enough to be its own class.

Comment (sjcorbett) [inline brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaPlanToSpecTransformer.java]: Also interesting enough to require at least basic tests.

File: brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaPlanToSpecTransformer.java [hunk 4]
```diff
@@ -184,79 +190,118 @@ private static boolean isTosca(Map<?,?> obj) {
...
+        public DependencyTree(Topology topo, ManagementContext mgmt, CSARRepositorySearchService repositorySearchService, CsarFileRepository csarFileRepository) {
+            this.topo = topo;
+            this.repositorySearchService = repositorySearchService;
+            this.csarFileRepository = csarFileRepository;
+            this.nodeTemplates = topo.getNodeTemplates();
+            this.mgmt = mgmt;
+            for(Entry<String, NodeTemplate> nodeTemplate : nodeTemplates.entrySet()){
+                String parentId = getParentId(nodeTemplate.getValue());
+                parents.put(nodeTemplate.getKey(), parentId);
+                children.putIfAbsent(parentId, MutableList.<String>of());
```
Comment (sjcorbett) [inline brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaPlanToSpecTransformer.java]: `putIfAbsent` also Java 1.8 on.

**Comment by robertgmoss** [comment]
@sjcorbett comments addressed

**Comment by sjcorbett** [comment]
@robertgmoss Thanks. Care to squash and merge?

**Pull request merged**